### PR TITLE
[PVR] Fix channel playback time issues.

### PR DESCRIPTION
**Pull request by ksooo**
@FernetMenta does this fix the issues you reported? => https://github.com/xbmc/xbmc/commit/c833ef5a43718c8766b01f2dd93c755e3e3f342b#r39397914 ff.

### CODE REVIEW

**Comment by FernetMenta** [comment]
thx! I will try the fix this evening and check tomorrow morning if channel epg look better.

**Comment by FernetMenta** [comment]
This is what I observed this morning:

- power on, channels view visible
- epg tags for channels were correct (current time)
- after about 10 seconds the system recognized that the backend was not available -> fly in note
- all channels disappeared. I haven't seen this behaviour so far but I like it because there were actually no services available. would be nice if the system checked the backend immediately after wakeup
- after the backed had become available, the channels appeared with correct epg

**Comment by ksooo** [comment]
Thanks for the feedback. It seems, that the issues with playback time after resume are fixed.

Regarding the delay: Not 100% sure, but this could just be the ressponse timeout addon->tvh server as the addon already tries to reconnect to the server immediately when addon's OnResume function gets called by Kodi, which is pretty early during resume. 

@phunkyfish mind taking a look at the code change? Will going to merge the PR when you are fine.

**Comment by FernetMenta** [comment]
> but this could just be the ressponse timeout addon->tvh server

probably, but right after wakeup the connection state should be disconnected.

File: xbmc/pvr/epg/EpgInfoTag.cpp [hunk 1]
```diff
@@ -196,12 +196,8 @@ int CPVREpgInfoTag::ClientID() const
...
 CDateTime CPVREpgInfoTag::GetCurrentPlayingTime() const
 {
-  const std::shared_ptr<CPVRPlaybackState> playbackState =
-      CServiceBroker::GetPVRManager().PlaybackState();
-  if (playbackState && playbackState->IsPlayingChannel(ClientID(), UniqueChannelID()))
-    return playbackState->GetPlaybackTime();
-  else
-    return CDateTime::GetUTCDateTime();
+  return CServiceBroker::GetPVRManager().PlaybackState()->GetChannelPlaybackTime(ClientID(),
+                                                                                 UniqueChannelID());
```
Comment (phunkyfish) [inline xbmc/pvr/epg/EpgInfoTag.cpp:200]: Long line I think.

**Review by phunkyfish — APPROVED** [review]
Just a minor on line length. Nice

File: xbmc/pvr/epg/EpgInfoTag.cpp (hunk 1 above)
Comment (ksooo) [reply xbmc/pvr/epg/EpgInfoTag.cpp:200]: Nope. clang-format is happy as the line length is EXACTLY 100.

Comment (phunkyfish) [reply xbmc/pvr/epg/EpgInfoTag.cpp:200]: My bad.

Comment (ksooo) [reply xbmc/pvr/epg/EpgInfoTag.cpp:200]: np

**Comment by FernetMenta** [comment]
I had been watching a channel for an hour or so. Back in channels view a lot of channels disappeared. I haven't noticed this before. Maybe related to this change.

**Comment by FernetMenta** [comment]
Disregard, something else was wrong. Sorry.

**Comment by Leatherface75** [comment]
> > but this could just be the ressponse timeout addon->tvh server
> 
> probably, but right after wakeup the connection state should be disconnected.

I don't agree everytime i loose connection on Wifi i have to to wait a few minutes before channels is working again here and after a wakeup on my main HTPC even on ethernet connection it's a mess.
I have to restart Kodi often for it work normallt again.
I don't have that behaviour on Android apps, Dreamboxes etc and i can open everything directly and don't need to wait. Atleast there should be an option to not timeout connection when loosing connection on Wifi or wakeup again. It should only give an error if it can't open the connection again.

**Pull request merged**